pkg/hintrunner/zero/zerohint_blake.go

```diff
@@ -0,0 +1,85 @@
+package zero
+
+import (
+	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
+	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	mem "github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
+	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
+	"math"
+	"math/big"
+)
+
+func newBlake2sAddUint256BigendHint(low, high, data hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "Blake2sAddUint256Bigend",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> B = 32
+			//> MASK = 2 ** 32 - 1
+			//> segments.write_arg(ids.data, [(ids.high >> (B * (3 - i))) & MASK for i in range(4)])
+			//> segments.write_arg(ids.data + 4, [(ids.low >> (B * (3 - i))) & MASK for i in range(4)])
+
+			low, err := hinter.ResolveAsFelt(vm, low)
+			if err != nil {
+				return err
+			}
+			high, err := hinter.ResolveAsFelt(vm, high)
+			if err != nil {
+				return err
+			}
+			dataPtr, err := hinter.ResolveAsAddress(vm, data)
+			if err != nil {
+				return err
+			}
+
+			var lowBig big.Int
+			var highBig big.Int
+			low.BigInt(&lowBig)
+			high.BigInt(&highBig)
+
+			const b uint64 = 32
+			mask := new(big.Int).SetUint64(math.MaxUint32)
+
+			for i := uint64(0); i < 4; i++ {
+				shift := uint(b * (3 - i))
+
+				highResultBig := new(big.Int).Set(&highBig)
+				// TODO: check if this mod is even needed
+				highResultBig.Rsh(highResultBig, shift).And(highResultBig, mask).Mod(highResultBig, fp.Modulus())
```

Comment:
I dont think that this modulus is needed, as the maximum value generated by this is highResultBig with no shift, which still can hold maximum of PRIME - 1